hamilflow/models/harmonic_oscillator_chain.py

```diff
@@ -0,0 +1,142 @@
+from functools import cached_property
+from typing import Mapping, Sequence, cast
+
+import numpy as np
+import pandas as pd
+from numpy.typing import ArrayLike
+from scipy.fft import ifft
+
+from .free_particle import FreeParticle
+from .harmonic_oscillator import ComplexSimpleHarmonicOscillator
+
+
+class HarmonicOscillatorsChain:
+    r"""Generate time series data for a coupled harmonic oscillator chain
+    with periodic boundary condition.
+
+    A one-dimensional circle of $N$ interacting harmonic oscillators can be described by the Lagrangian action
+    $$S_L[x_i] = \int_{t_0}^{t_1}\mathbb{d} t \left\\{ \sum_{i=0}^{N-1} \frac{1}{2}m \dot x_i^2 - \frac{1}{2}m\omega^2\left(x_i - x_{i+1}\right)^2 \right\\}\\,,$$
+    where $x_N \coloneqq x_0$.
+
+    This system can be solved in terms of _travelling waves_, obtained by discrete Fourier transform.
+
+    Since the original degrees of freedom are real, the initial conditions of the propagating waves need to satisfy
+    $Y_k = Y^*_{-k \mod N}$, see [Wikipedia](https://en.wikipedia.org/wiki/Discrete_Fourier_transform#DFT_of_real_and_purely_imaginary_signals).
```

Comment:
I think it is a bit easier for our future selves if we have have a usage example in the docstring.
Something like

https://github.com/kausalflow/hamilflow/blob/04ea13c523f8beb7d48123754ec96251b98b5c49/hamilflow/models/harmonic_oscillator.py#L141-L151

Reply:
Issue created https://github.com/kausalflow/hamilflow/issues/73